### PR TITLE
[lastools] Update to 2.0.2, fix include directories

### PR DESCRIPTION
**Pull request by Obbart**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Obbart** [comment]
@microsoft-github-policy-service agree

**Comment by Obbart** [comment]
> @Obbart please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Review by instafirm2023 — DISMISSED** [review]

**Comment by Cheney-W** [comment]
x64-osx failure:
```
/Users/vagrant/Data/buildtrees/lastools/src/v2.0.2-cfbfed1eda.clean/src/lasindex.cpp:146:25: error: character too large for enclosing character literal type
      if (argv[i][0] == '�') argv[i][0] = '-';
```
log: https://dev.azure.com/vcpkg/public/_build/results?buildId=85696&view=artifacts&pathAsName=false&type=publishedArtifacts

File: ports/lastools/portfile.cmake [hunk 1]
```diff
@@ -7,19 +7,21 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO LAStools/LAStools
-    REF 7c444a4bbae16fe43c676824a26419bb740a6ab8
-    SHA512 4503b033a5319caee5570f25a05009c0d05a8c61a43ed78317899faaeba82fbc1f9a8c8433772bb36547c74e850a2e68d20a16f6f22b12b3004da7e1aa2f334b
+    REF v2.0.2
```
Comment (Cheney-W) [inline ports/lastools/portfile.cmake:10]:
```suggestion
    REF "v${VERSION}"
```

**Comment by Cheney-W** [comment]
Please remove the records of the lastools from https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt because `"supports": "!uwp"` has been added into the vcpkg.json file. 
```
lastools:arm-uwp=fail
lastools:x64-uwp=fail
```

File: ports/lastools/portfile.cmake (hunk 1 above)
Comment (Obbart) [reply ports/lastools/portfile.cmake:10]: To solve this: https://github.com/microsoft/vcpkg/pull/29770#issuecomment-1439465504 I had to update the REF to latest master commit

**Comment by Obbart** [comment]
> Please remove the records of the lastools from https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt because `"supports": "!uwp"` has been added into the vcpkg.json file.
> 
> ```
> lastools:arm-uwp=fail
> lastools:x64-uwp=fail
> ```

References removed

File: ports/lastools/vcpkg.json [hunk 2]
```diff
@@ -1,8 +1,18 @@
 {
   "name": "lastools",
-  "version-string": "2020-05-09",
+  "version": "2.0.2",
   "port-version": 1,
```
Comment (Cheney-W) [inline ports/lastools/vcpkg.json:4]:
```suggestion
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: versions/l-/lastools.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "145609edd1fd7d755acd3d12a26a9804e3e85a99",
+      "version": "2.0.2",
+      "port-version": 0
+    },
+    {
+      "git-tree": "a0ad6c35f7aa75b6619c20639c9f4fb9ac2fd1b6",
+      "version": "2.0.2",
+      "port-version": 1
+    },
```
Comment (Cheney-W) [inline versions/l-/lastools.json:12]:
```suggestion
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/lastools/vcpkg.json [hunk 4]
```diff
@@ -1,8 +1,17 @@
 {
   "name": "lastools",
-  "version-string": "2020-05-09",
-  "port-version": 1,
+  "version": "2.0.2",
```
Comment (dg0yt) [inline ports/lastools/vcpkg.json:3]: This claims a wrong version for the commit you are using.
I think you should use `2.0.2+20230206`.
https://github.com/LAStools/LAStools/compare/v2.0.2...2bbbfa9

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Review by dg0yt — APPROVED** [review]

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**